### PR TITLE
BUG 1796927: *: Move metrics whitelist to downstream usage

### PR DESCRIPTION
**Pull request by brancz**
This is continuing the change of having the metrics whitelist be defined not in the telemeter repo but in its downstream usages (the client in cluster-monitoring-operator, and the server in the production manifests which are synced from the cluster-monitoring-operator's manifests).

@LiliC @squat @metalmatze @kakkoyun

### CODE REVIEW

**Comment by brancz** [comment]
> Can you remove the https://github.com/openshift/telemeter#adding-new-metrics-to-send-via-telemeter and instead just leave the docs link.

Done.

> You need bugzilla for this as well :)

I don't think we do, as telemeter is a production service that ships out of openshift release cycle. I'm not 100% certain though, let's see if the bot ends up complaining.

**Comment by LiliC** [comment]
> tide Pending — Not mergeable. Needs bugzilla/valid-bug, lgtm labels.

See tide job ;) It applies to all repos in openshift org :)

File: metrics.json [hunk 1]
```diff
@@ -1 +0,0 @@
-jsonnet/telemeter/metrics.jsonnet
```
Comment (LiliC) [inline metrics.json]: Is this file still needed?

File: jsonnet/telemeter/benchmark.libsonnet [hunk 2]
```diff
@@ -6,7 +6,7 @@ local b = (import 'prometheus-operator/prometheus-operator.libsonnet') +
   _config+:: {
     namespace: 'telemeter-benchmark',
     telemeterServer+: {
-      whitelist+: (import 'metrics.jsonnet'),
+      whitelist+: [],
```
[the remark this answers is not in the document]
Comment (brancz) [reply jsonnet/telemeter/benchmark.libsonnet]: We just want to make sure there is an array there.

File: metrics.json (hunk 1 above)
Comment (brancz) [reply metrics.json]: Nope

Comment (LiliC) [reply metrics.json]: Why did you resolve the comment then? :)

File: jsonnet/telemeter/benchmark.libsonnet (hunk 2 above)
[the remark this answers is not in the document]
Comment (LiliC) [reply jsonnet/telemeter/benchmark.libsonnet]: I see, thanks.

[the remark this answers is not in the document]
Comment (brancz) [reply jsonnet/telemeter/benchmark.libsonnet]: Still needs a lgtm :)

[the remark this answers is not in the document]
Comment (LiliC) [reply jsonnet/telemeter/benchmark.libsonnet]: There is one comment you said that the file is not needed so we can remove it, no?

[the remark this answers is not in the document]
Comment (LiliC) [reply jsonnet/telemeter/benchmark.libsonnet]: nvm github being github showed me like the file was not deleted.
